clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1013,16 +1013,8 @@ private void maybeFailWithError() {
         if (!hasError()) {
             return;
         }
-        // for ProducerFencedException, do not wrap it as a KafkaException
-        // but create a new instance without the call trace since it was not thrown because of the current call
-        if (lastError instanceof ProducerFencedException) {
-            throw new ProducerFencedException("Producer with transactionalId '" + transactionalId
-                    + "' and " + producerIdAndEpoch + " has been fenced by another producer " +
-                    "with the same transactionalId");
-        }
-        if (lastError instanceof InvalidProducerEpochException) {
```

Comment:
I saw we had some tests that verified producer fenced, but do we not have any for invalid producer epoch here?

Reply:
`INVALID_PRODUCER_EPOCH` is always converted to a `ProducerFencedException`, so we should not see `InvalidProducerEpochException` here anymore, right? There are tests mocking a `INVALID_PRODUCER_EPOCH` response though.